lib/src/config/config.rs

```
@@ -855,10 +805,10 @@ impl ClientConfigBuilder {
         #[cfg(feature = "metrics-server")]
         {
             if let Some(metrics_config) = &config_file.metrics_server {
-                let ip = metrics_config
-                    .bind
-                    .as_ref()
-                    .and_then(|addr| addr.into_ip_address());
+                let ip = metrics_config.bind.as_ref().map(|addr| {
+                    addr.parse()
+                        .expect("Error parsing the IP address of the metrics server")
```

I'd prefer to return a graceful error